c7n/resources/sns.py

```diff
@@ -468,3 +468,63 @@ def process(self, resources):
                 client.delete_topic(TopicArn=r['TopicArn'])
             except client.exceptions.NotFoundException:
                 continue
+
+
+@resources.register('sns-subscription')
+class SNSSubscription(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'sns'
+        enum_spec = ('list_subscriptions', 'Subscriptions', None)
+        id = name = dimension = 'SubscriptionArn'
+        arn = 'SubscriptionArn'
+        cfn_type = 'AWS::SNS::Subscription'
+        default_report_fields = (
+            'SubscriptionArn',
+            'Owner',
+            'Protocol',
+            'Endpoint',
+            'TopicArn'
+        )
+
+
+@SNSSubscription.action_registry.register('delete')
```

Comment:
I think we can change `delete` to a better action name since its not deleting

Reply:
Yea, I wasn't sure about this one. It is technically deleting the subscription, but not the endpoint. I was going to use 'unsubscribe' but I thought since we were moving toward more uniform action names that 'delete' was better. Do you think I should change it to 'unsubscribe' still?

Reply:
@kapilt thoughts??

Reply:
i could go either way, i'm fine with either delete or unsubscribe.